modules/ztp-image-based-upgrade-creating-backup-resources.adoc

```diff
@@ -0,0 +1,270 @@
+====
+--
+
+. Generate a `ConfigMap` object for your OADP CRs.
+
+.. Create the `ConfigMap` object:
++
+[source,terminal]
+----
+$ oc create configmap example-oadp-cm --from-file=example-oadp-resources.yaml=<path_to_oadp_crs> -n openshift-adp
+----
++
+.Example output
+[source,terminal]
+----
+kind: ConfigMap
+metadata:
+  name: oadp-cm-example
+  namespace: openshift-adp
+[...]
+----
+
+.. Apply the `ConfigMap`:
++
+[source,terminal]
+----
+$ oc apply -f oadp-cm-example.yaml
+----
+
+. (Optional) To keep your custom catalog sources after the upgrade, generate another `ConfigMap` object for your catalog sources and add them to the `spec.extraManifest` in the `ImageBasedUpgrade` CR. For more information about catalog sources, see xref:https://access.redhat.com/documentation/en-us/openshift_container_platform/4.15/html-single/operators/index#olm-catalogsource_olm-understanding-olm[Catalog source].
+
+.. Create the `CatalogSource` CR.
++
+--
+[source,yaml]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  annotations:
+    target.workload.openshift.io/management: '{"effect": "PreferredDuringScheduling"}'
+  name: example-catalogsources
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  displayName: disconnected-redhat-operators
+  image: quay.io/example-org/example-catalog:v1
+----
+--
+
+.. Create the `ConfigMap` object:
++
+--
+[source,terminal]
+----
+$ oc create configmap example-extra-manifests-cm --from-file=example-catalogsources.yaml=<path_to_catalogsource_cr> -n openshift-lifecycle-agent
+----
+
+
+.Example output
+[source,terminal]
+----
+kind: ConfigMap
+metadata:
+  name: example-extra-manifests-cm
+  namespace: openshift-lifecycle-agent
+[...]
+----
+--
+
+. Edit the `ImageBasedUpgrade` CR:
++
+[source,yaml]
+----
+apiVersion: lca.openshift.io/v1alpha1
+kind: ImageBasedUpgrade
+metadata:
+  name: example-upgrade
+#  annotations:
+#    auto-rollback-on-failure.lca.openshift.io/post-reboot-config: Disabled <1>
+#    auto-rollback-on-failure.lca.openshift.io/upgrade-completion: Disabled <2>
+#    auto-rollback-on-failure.lca.openshift.io/init-monitor: Disabled <3>
+spec:
+  stage: Idle
+  seedImageRef:
+    version: 4.15.2 <4>
+    image: <seed_container_image> <5>
+    pullSecretRef: <seed_pull_secret> <6>
+  autoRollbackOnFailure: {} <7>
+#    initMonitorTimeoutSeconds: 1800 <8>
+  extraManifests: <9>
+  - name: example-extra-manifests
+    namespace: openshift-lifecycle-agent
+  oadpContent: <10>
+  - name: oadp-cm-example
+    namespace: openshift-adp
+----
+<1>	(Optional) Disables automatic rollback when the reconfiguration of the cluster fails upon the first reboot.
+<2> (Optional) Disables automatic rollback after the {lcao} reports a failed upgrade upon completion.
+<3> (Optional) Disables automatic rollback when the upgrade does not complete after reboot within the time frame specified in the `spec.autoRollbackOnFailure.initMonitorTimeoutSeconds` field.
```

Comment:
i dont think we want users to know how to disable this. The only thing they can change is `initMonitorTimeoutSeconds`. 

The disable feature is strictly for devs from here on out(we only note this config in [HACKING.md)](https://github.com/openshift-kni/lifecycle-agent/blob/main/HACKING.md)

Reply:
@pixelsoccupied So if they set the `initMonitorTimeoutSeconds`, it'll be automatically disabled?

I'll remove the annotations and their explanation from the official docs.

Reply:
So basically users must never disable the auto-rollback feature (only thing can change is the timeout/how long until rollback kicks in)

Reply:
Noted! In this case, I'm also removing this note "By default, automatic rollback on failure is enabled." for `autoRollbackOnFailure`